lib/rules/no-assignment-of-computed-property-dependencies.js

```diff
@@ -0,0 +1,191 @@
+  if (types.isClassDeclaration(nodeClass)) {
+    // Native class.
+    return javascriptUtils.flatMap(nodeClass.body.body, (node) => {
+      const computedDecorator = decoratorUtils.findDecorator(node, 'computed');
+      if (computedDecorator) {
+        return getComputedPropertyDependentKeys(computedDecorator.expression);
+      } else {
+        return [];
+      }
+    });
+  } else if (types.isCallExpression(nodeClass)) {
+    // Classic class.
+    if (
+      nodeClass.arguments &&
+      nodeClass.arguments.length > 0 &&
+      types.isObjectExpression(nodeClass.arguments[nodeClass.arguments.length - 1])
+    ) {
+      const classObject = nodeClass.arguments[nodeClass.arguments.length - 1];
+      return javascriptUtils.flatMap(classObject.properties, (node) => {
+        if (emberUtils.isComputedProp(node.value) && node.value.arguments) {
+          return getComputedPropertyDependentKeys(node.value);
+        } else {
+          return [];
+        }
+      });
+    }
+  } else {
+    assert(false, 'Unexpected node type for an Ember class.');
+  }
+
+  return [];
+}
+
+/**
+ * Gets a list of tracked properties used inside a class.
+ *
+ * @param {Node} nodeClass - Node for the Ember class
+ * @returns {String[]} - list of tracked properties used inside the class
+ */
+function findTrackedProperties(nodeClassDeclaration) {
+  return nodeClassDeclaration.body.body
+    .filter(
+      (node) =>
+        types.isClassProperty(node) &&
+        decoratorUtils.hasDecorator(node, 'tracked') &&
+        types.isIdentifier(node.key)
+    )
+    .map((node) => node.key.name);
+}
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description:
+        'disallow assignment of untracked properties that are used as computed property dependencies',
+      category: 'Computed Properties',
+      recommended: false,
+      url:
+        'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/no-assignment-of-computed-property-dependencies.md',
+    },
+    fixable: 'code',
+    schema: [],
+  },
+
+  ERROR_MESSAGE,
+
+  create(context) {
+    if (emberUtils.isTestFile(context.getFilename())) {
+      // This rule does not apply to test files.
+      return {};
+    }
+
+    // State being tracked for this file.
+    let trackedProperties = new Set();
+    let computedPropertyDependentKeys = new Set();
+    let currentEmberModule = null;
+
+    return {
+      ClassDeclaration(node) {
+        if (emberUtils.isAnyEmberModule(context, node)) {
+          currentEmberModule = node;
+
+          // Gather computed property dependent keys from this class.
+          computedPropertyDependentKeys = new Set(findComputedPropertyDependentKeys(node));
+
+          // Gather tracked properties from this class.
+          trackedProperties = new Set(findTrackedProperties(node));
+        }
+      },
+
+      CallExpression(node) {
+        if (emberUtils.isAnyEmberModule(context, node)) {
```

Comment:
The tracked properties need to be reset here.

Reply:
Fixed.